File: pinkmoe_server/logic/admin/topic.go
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-06-22 11:13:07
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 08:58:49
 * @FilePath: /pinkmoe_server/logic/admin/topic.go
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe 
 * 问题反馈qq群:749150798
 * xanaduCms程序上所有内容(包括但不限于 文字，图片，代码等)均为指针科技原创所有，采用请注意许可
 * 请遵循 “非商业用途” 协议。商业网站或未授权媒体不得复制内容，如需用于商业用途或者二开，请联系作者捐助任意金额即可，我们将保存所有权利。
 * Copyright (c) 2022 by 指针科技, All Rights Reserved.
 */
package adminLogic

import (
	"server/dao/mysql"
	"server/model"
	"server/model/request"
)

func GetTopicList(p request.SearchTopicParams) (err error, list interface{}, total int64) {
	err, list, total = mysql.GetTopicList(p)
	return
}

func CreateTopic(p model.XdTopic) (err error) {
	err = mysql.CreateTopic(p)
	return
}

func UpdateTopic(p model.XdTopic) (err error) {
	err = mysql.UpdateTopic(p)
	return
}

func DeleteTopic(p model.XdTopic) (err error) {
	err = mysql.DeleteTopic(p)
	return
}

File: pinkmoe_admin/src/api/content/topic.ts
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-05-06 14:55:21
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 09:14:09
 * @FilePath: /pinkmoe_admin/src/api/content/topic.ts
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe 
 * 问题反馈qq群:749150798
 * xanaduCms程序上所有内容(包括但不限于 文字，图片，代码等)均为指针科技原创所有，采用请注意许可
 * 请遵循 “非商业用途” 协议。商业网站或未授权媒体不得复制内容，如需用于商业用途或者二开，请联系作者捐助任意金额即可，我们将保存所有权利。
 * Copyright (c) 2022 by 指针科技, All Rights Reserved. 
 */
import { http } from '@/utils/http/axios';
import { BasicResponseModel } from '@/api/common';

/**
 * @description: 获取话题列表
 * @param params
 */
export function getTopicList(params?) {
  return http.request<BasicResponseModel>(
    {
      url: '/Admin/Topic/TopicList',
      method: 'GET',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 更新话题
 * @param params
 */
export function updateTopic(params) {
  return http.request(
    {
      url: `/Admin/Topic/TopicUpdate`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 创建话题
 * @param params
 */
export function createTopic(params) {
  return http.request(
    {
      url: `/Admin/Topic/TopicCreate`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 删除话题
 * @param params
 */
export function deleteTopic(params) {
  return http.request(
    {
      url: `/Admin/Topic/TopicDelete`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}
